plugin/group/scaled.go

```diff
@@ -38,34 +38,42 @@ func (s *scaledGroup) changeSettings(settings groupSettings) {
 	s.settings = settings
 }
 
-func (s *scaledGroup) CreateOne(logicalID *instance.LogicalID) {
+// latestSettings gives a point-in-time view of the settings for this group.  This allows other functions to
+// safely use settings and make calls to plugins without holding the lock.
+func (s *scaledGroup) latestSettings() groupSettings {
```

Comment:
I am not sure if this does much.  The lock is held as soon as the settings, which you want only for read, is copied and returned on the stack.   Ideally, you want to block reads only when an update of the settings is being updated.  So I think the `scaledGroup.lock` should be a `RWMutex`.  Then this function isn't needed and you just call `lock.RLock() / defer lock.RUnlock()`.   Concurrent reads won't block if no update on the setting is in progress (as signaled by `lock.Lock() / lock.Unlock()`).